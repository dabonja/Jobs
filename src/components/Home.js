import React from 'react';
import JobDescription from './JobDescription';
import Selection from './Selection';
import { useEffect, useState } from 'react';
import Title from './Title'

const Home = ({ categoriesList }) => {

    let catList = [...categoriesList];
 
    const [selectedJobs, setSelectedJobs] = useState([]);
    const [optionState, setOptionState] = useState('');
    const handleChange = (e) => {
        setOptionState(prevState => prevState = e.target.value)
    }

    useEffect(() => {
        let filteredJobs = catList.filter(job => job.category_name === optionState)
        console.log(optionState);
        setSelectedJobs([...filteredJobs])
        console.log(selectedJobs);
    }, [optionState])

    let jobTypes = catList.map(job => job.category_name)
    let categories = [...new Set(jobTypes)];

    return (
        <div className="align-items-center justify-content-center d-flex" style={{ position:'relative', top: '10rem' }}>
            <div className="container">
                <Title title="Unajmite firmu za odredjeni posao!" />
                <select className="form-select" aria-label="Default select example" onChange={handleChange} >

                    <option defaultValue="selected" >Izaberite trazeni posao...</option>
                    {
                        categories.map(jobType => (
                            <option key={jobType.id} value={jobType.category_name}>{jobType}</option>
                        ))
                    }

                </select>

                <div className="col"  align="center">
                   
                        {
                            selectedJobs && selectedJobs.map(job => (
                                <JobDescription key={job.id} category={job.category_name} description={job.description} name={job.companyName} contact={job.contact} location={job.location} />
                            ))
                        }
                </div>
            </div>

        </div>
    )
}

export default Home

/*
  <div className="list-inline">

            {
                    jobs && jobs.map( job =>(
                        <JobDescription key= {job.id} category = {job.category} description={job.description} name={job.firm} contact={job.contact} location={job.location} />
                    ))
                }

            </div>
*/