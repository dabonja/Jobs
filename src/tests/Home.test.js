import Home from '../components/Home';
import renderer from 'react-test-renderer';


it('renders Home component correctly', () => {

    let category = [
        { companyName: 'MetalIn', description: 'neki opis', location: 'Belgrade', contact: '9588390204', category_name: 'Moleraj' },
        { companyName: 'Malisa', description: 'neki opis', location: 'Novi Sad', contact: '9588390204', category_name: 'Bravar' },
        { companyName: 'SZR KAMACO', description: 'neki opis', location: 'Indjija', contact: '9588390204', category_name: 'Gradjevina\r\n' },
        { companyName: 'Firma1', description: 'neki opis', location: 'Zrenjanin', contact: '534234234232', category_name: 'Menadzer' }
    ]
    const tree = renderer.create(<Home categoriesList={category} />)
        .toJSON();
    expect(tree).toMatchSnapshot()
})