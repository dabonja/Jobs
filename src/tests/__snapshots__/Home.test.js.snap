// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component correctly 1`] = `
<div
  className="align-items-center justify-content-center d-flex"
  style={
    Object {
      "position": "relative",
      "top": "10rem",
    }
  }
>
  <div
    className="container"
  >
    <div>
      <h1>
        Hire a professional company!
      </h1>
    </div>
    <select
      aria-label="Default select example"
      className="form-select"
      onChange={[Function]}
    >
      <option
        defaultValue="selected"
      >
        Choose the right work for you...
      </option>
      <option>
        Moleraj
      </option>
      <option>
        Bravar
      </option>
      <option>
        Gradjevina

      </option>
      <option>
        Menadzer
      </option>
    </select>
    <div
      align="center"
      className="col"
    >
      <h1
        style={
          Object {
            "display": "none",
          }
        }
      >
        There are currently no registered companies for the selected job type.
      </h1>
    </div>
  </div>
</div>
`;
